airbyte-webapp/src/packages/cloud/components/experiments/FreeConnectorProgram/hooks/useFreeConnectorProgram.tsx

```diff
@@ -0,0 +1,33 @@
+import { ReleaseStage, WebBackendConnectionRead } from "core/request/AirbyteClient";
+import { useExperiment } from "hooks/services/Experiment";
+import { useDestinationDefinitionList } from "services/connector/DestinationDefinitionService";
+import { useSourceDefinitionList } from "services/connector/SourceDefinitionService";
+
+export const useFreeConnectorProgram = () => {
+  const isFreeConnectorProgramEnabled = useExperiment("workspace.freeConnectorsProgram.visible", false);
+
+  const { sourceDefinitions } = useSourceDefinitionList();
+  const { destinationDefinitions } = useDestinationDefinitionList();
```

Comment:
We don't need to fetch and iterate through the connector definitions in the frontend, the backend will pass us a boolean for that alongside enrollment status: https://github.com/airbytehq/airbyte/blob/5ce541f28c8edbdd963d549994dbf5cb14bcba60/airbyte-webapp/src/packages/cloud/lib/domain/freeConnectorProgram/api.ts#L38-L39

Given that, I'd prefer to remove the calls to `use{Source,Destination}DefinitionList` and just hardcode a boolean response inside `connectionHasAlphaOrBetaConnector` instead, so we don't make any extra network calls when real users load the connection page.

Reply:
Ooh amazing! I missed that part of the response. I was very much hoping to not need these calls.

Reply:
If the user has _any_ free connectors, we want to show the enrollment banners, even if the one they're currently viewing is GA->GA. The banner text doesn't specify that the _current_ connector is eligible, anyway.

Reply:
I see.  Cool.  Will tear that out then!